### DISCUSSION TITLE
Can Add ChunkBy method for Enumerable?

### DISCUSSION THREAD

**Discussion by zms9110750** [dotnet/csharplang#6033] [General] [answered]
I'm working on a translation program. The API I use can accept up to 5000 characters.
Of course, for translation, you can't cut the text from the middle.
Therefore, the ChunkBy method should be able to a specified key selector function, trace back to the previous element, and split it as a boundary.

**Comment by theunrepentantgeek** [comment, marked as answer]
This repo is for discussing changes to the C# language itself, not the libraries that make up .Net, so I'm afraid you've come to the wrong place. 

I think the [MoreLinq](https://morelinq.github.io) project has what you need however - check out the `Segment()` method.

**Reply by zms9110750** [reply]
Oh, sorry. I found this discussion area from the F1 help document of vs. Can you tell me where I should go?

**Reply by Joe4evr** [reply]
API proposals should be made in dotnet/runtime.
